pyxrf/model/draw_image_rgb.py

```diff
@@ -323,6 +328,52 @@ def show_image(self):
         except IndexError:
             selected_data = np.ones([3,10,10])
 
+
+        def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
+            """
+            Compute ranges for x- and y- axes of the plot. Make sure that the ranges for x- and y-axes are
+            always equal and fit the maximum of the ranges for x and y values:
+                  max(abs(x_max-x_min), abs(y_max-y_min))
+            The ranges are set so that the data is always centered in the middle of the ranges
+
+            Parameters
+            ----------
+
+            x_min, x_max, y_min, y_max : float
+                lower and upper boundaries of the x and y values
+
+            Returns
+            -------
+
+            x_axis_min, x_axis_max, y_axis_min, y_axis_max : float
+                lower and upper boundaries of the x- and y-axes ranges
+            """
+
+            x_axis_min, x_axis_max, y_axis_min, y_axis_max = x_min, x_max, y_min, y_max
+            x_range, y_range = abs(x_max - x_min), abs(y_max - y_min)
+            if x_range > y_range:
+                y_center = (y_max + y_min) / 2
+                y_axis_max = y_center + x_range / 2
+                y_axis_min = y_center - x_range / 2
+            else:
+                x_center = (x_max + x_min) / 2
+                x_axis_max = x_center + y_range / 2
+                x_axis_min = x_center - y_range / 2
+
+            return x_axis_min, x_axis_max, y_axis_min, y_axis_max
```

Comment:
This can potentially be deduplicated, as there is one more similar block of code above.